### PR TITLE
Add --dry-run to teuthology-nuke

### PR DESCRIPTION
**Pull request by zmc**
Signed-off-by: Zack Cerza <zack@redhat.com>

### CODE REVIEW

File: scripts/nuke.py [hunk 1]
```diff
@@ -19,6 +20,7 @@
                         archive path for a job to kill and nuke
   --stale               attempt to find and nuke 'stale' machines
                         (e.g. locked by jobs that are no longer running)
+  --dry-run             Don't actually nuke anything
```
Comment (andrewschoen) [inline scripts/nuke.py]: So what does --dry-run actually do then except make it a noop.  Does it have any output? I guess, what's the use case here?

Comment (zmc) [inline scripts/nuke.py]: It still outputs the list of targets that would be nuked.

Comment (andrewschoen) [inline scripts/nuke.py]: Ok, can we change the help text to mention that then? "Don't actually nuke anything, but output the list of targets to be nuked". Maybe?

Comment (zmc) [inline scripts/nuke.py]: Of course! I'll do that.

**Pull request merged**